hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -279,15 +279,37 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 	// Set Ignition Server endpoint
 	{
+		serviceStrategy := servicePublishingStrategyByType(hcluster, hyperv1.Ignition)
+		if serviceStrategy == nil {
+			return ctrl.Result{}, fmt.Errorf("ignition server service strategy not specified")
+		}
 		controlPlaneNamespace := manifests.HostedControlPlaneNamespace(hcluster.Namespace, hcluster.Name)
-		ignitionServerRoute := ignitionserver.Route(controlPlaneNamespace.GetName())
-		if err := r.Client.Get(ctx, client.ObjectKeyFromObject(ignitionServerRoute), ignitionServerRoute); err != nil {
-			if !apierrors.IsNotFound(err) {
-				return ctrl.Result{}, fmt.Errorf("failed to get ignitionServerRoute: %w", err)
+		switch serviceStrategy.Type {
+		case hyperv1.Route:
+			ignitionServerRoute := ignitionserver.Route(controlPlaneNamespace.GetName())
+			if err := r.Client.Get(ctx, client.ObjectKeyFromObject(ignitionServerRoute), ignitionServerRoute); err != nil {
+				if !apierrors.IsNotFound(err) {
+					return ctrl.Result{}, fmt.Errorf("failed to get ignitionServerRoute: %w", err)
+				}
 			}
-		}
-		if err == nil && ignitionServerRoute.Spec.Host != "" {
-			hcluster.Status.IgnitionEndpoint = ignitionServerRoute.Spec.Host
+			if err == nil && ignitionServerRoute.Spec.Host != "" {
+				hcluster.Status.IgnitionEndpoint = ignitionServerRoute.Spec.Host
+			}
+		case hyperv1.NodePort:
+			if serviceStrategy.NodePort == nil {
```

Comment:
in addition to logs should we also set a condition that signal this in a more ux friendly manner?

I wonder should we log the error but not return it here? since this won't get fixed by requeueing unless there's and update to the API which will already emit an update event a reconciliation will happen.

Reply:
I think we need to return error as it doesn't make sense to continue execution if this has failed. However I could also log if that helps.

Reply:
fixing agreed

Reply:
fixed